### PR TITLE
Extend basic_writer with write() method for doubles with optional format_specs argument

### PR DESCRIPTION
**Pull request by chronoxor**
I agree that my contributions are licensed under the {fmt} license, and agree to future changes to the licensing.

### CODE REVIEW

**Comment by chronoxor** [comment]
This is important for me, because I want to write doubles using basic_writer with format_specs in a same way I can write ints.

File: include/fmt/format.h [hunk 1]
```diff
@@ -2703,15 +2703,19 @@ template <typename Range> class basic_writer {
     write_int(value, s);
   }
 
-  void write(double value) { write_double(value, format_specs()); }
+  void write(double value, const format_specs& spec = format_specs()) {
```
Comment (vitaut) [inline include/fmt/format.h]: Can `format_specs()` be replaced with `{}` or will gcc 4.4 complain?

Comment (chronoxor) [reply include/fmt/format.h]: Fixed. This compiles with gcc 4.4

Comment (vitaut) [reply include/fmt/format.h]: I meant dropping the type name altogether and just using braces as in ` = {}`, not to repeat the argument type again.

Comment (chronoxor) [reply include/fmt/format.h]: Sorry for misunderstanding. Unfortunately `= {}` not works for gcc 4.4 so lets currently keep the solution with `= format_specs()`

**Comment by chronoxor** [comment]
Does the PR require any other fixes from me?

**Comment by vitaut** [comment]
> Does the PR require any other fixes from me?

Nope, all good. Merged and thanks for the PR!

**Comment by chronoxor** [comment]
Thanks a lot!

**Pull request merged**